src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInNameCheck.java

```diff
@@ -363,22 +362,46 @@ else if (abbrStarted) {
     /**
      * Get Abbreviation if it is illegal.
      * @param str name
-     * @param beginIndex begin index
-     * @param endIndex end index
-     * @return true is abbreviation is bigger that required and not in ignore list
+     * @param beginIndex begin index, inclusive
+     * @param endIndex end index, inclusive
+     * @return the abbreviation if bigger than required and not in ignore list
      */
     private String getAbbreviationIfIllegal(String str, int beginIndex, int endIndex) {
         String result = null;
         final int abbrLength = endIndex - beginIndex;
         if (abbrLength > allowedAbbreviationLength) {
-            final String abbr = str.substring(beginIndex, endIndex);
+            final String abbr = getAbbreviation(str, beginIndex, endIndex);
             if (!allowedAbbreviations.contains(abbr)) {
                 result = abbr;
             }
         }
         return result;
     }
 
+    /**
+     * Gets the abbreviation, only including the character at {@code endIndex} if
+     * it is the last character in the string, as it is usually the first capital in
+     * the next word.
+     * <p>
+     * For example, {@code getAbbreviation("getXMLParser", 3, 6)} returns "XML"
+     * (not "XMLP"), and so does {@code getAbbreviation("parseXML", 5, 7)}.
+     * </p>
+     * @param str name
+     * @param beginIndex begin index, inclusive
+     * @param endIndex end index, inclusive
```

Comment:
`inclusive` for `endIndex` is wrong.
It is only inclusive if `endIndex == str.length() - 1` is true.
Lets just drop the wording `inclusive` and add a comment somewhere that it isn't inclusive because `last capital is usually beginning of next word`, like what was used for the -1 wording.

Reply:
Good point, I think this can be clearer: `beginIndex` and `endIndex` are inclusive in the sense that `str.substring(beginIndex, endIndex + 1)` is a sequence of upper-case characters ending at / including `str.charAt(endIndex)` -- but `endIndex` is usually _not_ inclusive as far as what this method considers the actual abbreviation.